contracts/CarbonVoteXCore.sol

```diff
@@ -0,0 +1,368 @@
+    function setReceiver(
+        bytes32 namespace, 
+        address _receiverAddr, 
+        bytes32[] restrictedFunctions
+    ) 
+        public 
+        auth 
+    {
+        // prevent accidentally overwrite other receivers.
+        require(receiverAddr[namespace] == 0x0);
+
+        receiverAddr[namespace] = _receiverAddr;
+        setPermissions(restrictedFunctions, _receiverAddr);
+
+        emit _SetReceiver(namespace, _receiverAddr);
+    }
+
+    // @param namespace the namespace of CarbonVoteX receiver
+    // Remove receiver core and revoke all its permissions
+    // This function can only be called by root account (contract creator)
+    function cancelReceiver(bytes32 namespace) public auth {
+        // saving gas by prevent meaningless tx
+        require(receiverAddr[namespace] != 0x0);
+
+        address _receiverAddr = receiverAddr[namespace];
+        receiverAddr[namespace] = 0x0;
+
+        cancelPermissions(_receiverAddr);
+
+        emit _CancelReceiver(namespace, _receiverAddr);
+    }
+
+
+    // @param namespace the namespace of CarbonVoteX receiver
+    // @param startBlock starting block (unix timestamp) of the event
+    // @param endBlock ending block (unix timestamp) of the event
+    // @param pollId UUID (hash value) of a poll
+    // @param tokenAddr the address of the token
+    // a restricted function
+    // Register a new poll
+    // Note that we do not allow re-registrations for the same UUID
+    function register (
+        bytes32 namespace,
+        uint startBlock,
+        uint endBlock,
+        bytes32 pollId,
+        address tokenAddr
+    )
+        public
+        auth
+    {
+        // Check if poll exists.
+        require (!pollExist(namespace, pollId));
+        // check resonable endBlock and startBlock
+        require (startBlock < endBlock);
+
+        // Create a new poll and map the poll hashed value to the poll.
+        Poll memory poll = Poll({ 
+            startBlock: startBlock,
+            endBlock: endBlock,
+            pollId: pollId,
+            token: ERC20(tokenAddr)
+        });
+        polls[namespace][pollId] = poll;
+
+        emit _Register(msg.sender, namespace, pollId, startBlock, endBlock, tokenAddr);
+    }
+
+    // @param namespace the namespace of CarbonVoteX receiver 
+    // @param pollId UUID (hash value) of a poll
+    // @param voter address of a voter    
+    // @param votes number of votes to write
+    // a restricted function that can only be called by
+    // the master address and contract creator
+    function writeAvailableVotes(
+        bytes32 namespace, 
+        bytes32 pollId, 
+        address voter, 
+        uint votes
+    ) 
+        public
+        auth 
+    {
+        // poll must exit and has not yet expired. 
+        require (pollExist(namespace, pollId) && !pollExpired(namespace, pollId));
+        // a voter must not get vote twice.
+        require (!voteObtained(namespace, pollId, voter));
```

Comment:
remove  this condition: require (!voteObtained(namespace, pollId, voter));

Talked with Tim.  If we accidentally  wrote wrong  number of votes, we could still restore it.